Mage.Sets/src/mage/cards/b/BlazingSunsteel.java

```diff
@@ -111,11 +134,13 @@ public BlazingSunsteelEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source) {
-        Permanent creature = (Permanent) getValue("equipped");
-        Integer damage = (Integer) getValue("damage");
-        if (creature == null || damage == null || damage < 1) {
+        Permanent creature = game.getPermanentOrLKIBattlefield((UUID) getValue("equipped"));
+        int damage = (int)getValue("damage");
```

Comment:
Best practice to store and get values: read/write it as object like `Integer`. It will be compatible with `null` value (e.g. if you read a value before setup then you will not catch runtime error).

Reply:
Understood. In the previous PR I was told to use the primitive type, I'll swap it.

Reply:
I said to use the primitive int when you were just getting the power of a creature.  `getPower().getValue()` returns an int so it can never be null.  When you're saving and retrieving values like this, it's better to use Integer so you can null check (these methods return "Object" type.)